### ISSUE TITLE
detekt: Enable Comments.UndocumentedPublicFunction

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#77]
Libraries should have javadoc comments for all public functions. Detekt has a check for that.

In order to enable this we'd need add comments to those functions:

```
UndocumentedPublicFunction - [emptyResult] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:70:13
UndocumentedPublicFunction - [setOnCommitListener] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:75:5
UndocumentedPublicFunction - [setOnFilterListener] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:78:5
UndocumentedPublicFunction - [setOnSearchStateChangeListener] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:81:5
UndocumentedPublicFunction - [setOnTextChangeListener] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:84:5
UndocumentedPublicFunction - [setOnKeyPreImeListener] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:87:5
UndocumentedPublicFunction - [setOnSelectionChangedListener] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:90:5
UndocumentedPublicFunction - [setOnWindowsFocusChangeListener] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:93:5
UndocumentedPublicFunction - [<UnknownName>] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:122:31
UndocumentedPublicFunction - [<UnknownName>] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:146:25
UndocumentedPublicFunction - [<UnknownName>] at components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt:162:38
UndocumentedPublicFunction - [toSearchEngine] at components/browser/search/src/main/java/mozilla/components/browser/search/SearchEngineParser.kt:35:9
UndocumentedPublicFunction - [load] at components/browser/search/src/main/java/mozilla/components/browser/search/SearchEngineParser.kt:44:5
UndocumentedPublicFunction - [load] at components/browser/search/src/main/java/mozilla/components/browser/search/SearchEngineManager.kt:20:5
UndocumentedPublicFunction - [onValueAdded] at components/browser/session/src/main/java/mozilla/components/browser/session/ObservableList.kt:93:9
UndocumentedPublicFunction - [onValueRemoved] at components/browser/session/src/main/java/mozilla/components/browser/session/ObservableList.kt:94:9
UndocumentedPublicFunction - [onValueSelected] at components/browser/session/src/main/java/mozilla/components/browser/session/ObservableList.kt:95:9
UndocumentedPublicFunction - [onUrlChanged] at components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt:17:9
UndocumentedPublicFunction - [postToBackgroundThread] at components/support/utils/src/main/java/mozilla/components/support/utils/ThreadUtils.kt:18:5
UndocumentedPublicFunction - [postToMainThread] at components/support/utils/src/main/java/mozilla/components/support/utils/ThreadUtils.kt:22:5
UndocumentedPublicFunction - [postToMainThreadDelayed] at components/support/utils/src/main/java/mozilla/components/support/utils/ThreadUtils.kt:26:5
UndocumentedPublicFunction - [assertOnUiThread] at components/support/utils/src/main/java/mozilla/components/support/utils/ThreadUtils.kt:30:5
UndocumentedPublicFunction - [getString] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeBundle.kt:20:5
UndocumentedPublicFunction - [getParcelable] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeBundle.kt:32:5
UndocumentedPublicFunction - [hasExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:86:5
UndocumentedPublicFunction - [getBooleanExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:98:5
UndocumentedPublicFunction - [getIntExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:110:5
UndocumentedPublicFunction - [getStringExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:122:5
UndocumentedPublicFunction - [getBundleExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:134:5
UndocumentedPublicFunction - [getCharSequenceExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:151:5
UndocumentedPublicFunction - [getParcelableExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:163:5
UndocumentedPublicFunction - [getParcelableArrayListExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:175:5
UndocumentedPublicFunction - [getStringArrayListExtra] at components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt:187:5
UndocumentedPublicFunction - [onLocationChange] at components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt:19:9
```

**Comment by dector** [comment]
@pocmo Can we enable this rule now and include tech dept into the baseline file?

**Comment by pocmo** [comment]
OK 👍

**Comment by dector** [comment]
Enabled rule and updated baseline file. I guess issue title can be updated.

**Comment by NotWoods** [comment]
Seems that this was fixed by #4509
